EventFilter/Phase2PixelRawToDigi/plugins/Phase2ITQCoreProducer.cc

```diff
@@ -0,0 +1,226 @@
+  // Spacing 1 in column and 5 in row is introduced for each chip in between
+  // if neighboring chip exists
+  const int kQCoresInChipRow = (672);
+  const int kQCoresInChipColumn = (216);
+  const int kQCoresInChipRowGap = (5);
+  const int kQCoresInChipColumnGap = (10);
+}  // namespace
+
+Phase2ITDigiHit updateHitCoordinatesForLargePixels(Phase2ITDigiHit& hit) {
+  /*
+        In-place modification of Hit coordinates to take into account large pixels
+        Hits corresponding to large pixels are remapped so they lie on the boundary of the chip
+        Note that this operation can produce multiple hits with the same row/column coordinates
+        Duplicates get removed later on
+    */
+
+  // Current values before remapping
+  int row = hit.row();
+  int col = hit.col();
+
+  // Values after remapping
+  int updated_row = 0;
+  int updated_col = 0;
+
+  // Remapping of the row coordinate
+  if (row < kQCoresInChipRow) {
+    updated_row = row;
+  } else if (row < (kQCoresInChipRow + kQCoresInChipRowGap)) {
+    updated_row = kQCoresInChipRow - 1;
+  }  // This will be ignored for 2 chips module
+  else if (row < (kQCoresInChipRow + 2 * kQCoresInChipRowGap)) {
+    updated_row = kQCoresInChipRow;
+  } else {
+    updated_row = (hit.row() - 2 * kQCoresInChipRowGap);
+  }
+
+  // Remapping of the column coordinate
+  if (col < kQCoresInChipColumn) {
+    updated_col = col;
+  } else if (col < kQCoresInChipColumn + kQCoresInChipColumnGap) {
+    updated_col = kQCoresInChipColumn - kQCoresInChipColumnGap;
+  } else if (col < (kQCoresInChipColumn + 2 * kQCoresInChipColumn)) {
+    updated_col = kQCoresInChipColumn;
+  } else {
+    updated_col = (hit.col() - 2 * kQCoresInChipColumnGap);
+  }
+
+  hit.set_row(updated_row);
+  hit.set_col(updated_col);
+
+  return hit;
+}
+
+std::vector<Phase2ITDigiHit> adjustEdges(std::vector<Phase2ITDigiHit> hitList) {
+  /*
+        In-place modification of Hit coordinates to take into account large pixels
+    */
+  std::for_each(hitList.begin(), hitList.end(), &updateHitCoordinatesForLargePixels);
+  return hitList;
+}
+
+std::vector<Phase2ITChip> splitByChip(std::vector<Phase2ITDigiHit> hitList) {
+  // Split the hit list by read out chip
+  std::vector<std::vector<Phase2ITDigiHit>> hits_per_chip(4);
+  for (auto hit : hitList) {
+    int chip_index = (hit.col() < kQCoresInChipColumn) ? 0 : 1;
+    if (hit.row() >= kQCoresInChipRow) {
+      chip_index += 2;
+    }
+    hits_per_chip[chip_index].push_back(hit);
+  }
+
+  // Generate Phase2ITChip objects from the hit lists
+  std::vector<Phase2ITChip> chips;
+  for (int chip_index = 0; chip_index < 4; chip_index++) {
+    chips.push_back(Phase2ITChip(chip_index, hits_per_chip[chip_index]));
+  }
+
+  return chips;
+}
+
+std::vector<Phase2ITChip> processHits(std::vector<Phase2ITDigiHit> hitList) {
+  std::vector<Phase2ITDigiHit> newHitList;
+  newHitList = adjustEdges(hitList);
+
+  std::vector<Phase2ITChip> chips = splitByChip(newHitList);
```

Comment:
With my earlier comments this would become
```suggestion
  adjustEdges(hitList);
  std::vector<Phase2ITChip> chips = splitByChip(hitList);
```